pkg/controller/component/vars.go

```diff
@@ -594,19 +563,29 @@ func resolveServiceHostRef(ctx context.Context, cli client.Reader, synthesizedCo
 func resolveServicePortRef(ctx context.Context, cli client.Reader, synthesizedComp *SynthesizedComponent,
 	defineKey string, selector appsv1alpha1.ServiceVarSelector) ([]*corev1.EnvVar, []*corev1.EnvVar, error) {
 	resolvePort := func(obj any) (*corev1.EnvVar, *corev1.EnvVar) {
-		svc := obj.(*corev1.Service)
-		for _, svcPort := range svc.Spec.Ports {
-			if svcPort.Name == selector.Port.Name {
-				return &corev1.EnvVar{
-					Name:  defineKey,
-					Value: strconv.Itoa(int(svcPort.Port)),
-				}, nil
+		services := obj.([]*corev1.Service)
+		svcPorts := make(map[string]string)
+		for _, svc := range services {
+			for _, svcPort := range svc.Spec.Ports {
+				if svcPort.Name == selector.Port.Name {
+					svcPorts[svc.Name] = strconv.Itoa(int(svcPort.Port))
+					break
+				}
+			}
+			if len(svc.Spec.Ports) == 1 && (len(svc.Spec.Ports[0].Name) == 0 || len(selector.Port.Name) == 0) {
+				svcPorts[svc.Name] = strconv.Itoa(int(svc.Spec.Ports[0].Port))
 			}
 		}
-		if len(svc.Spec.Ports) == 1 && (len(svc.Spec.Ports[0].Name) == 0 || len(selector.Port.Name) == 0) {
+		if len(svcPorts) > 0 {
+			svcNames := maps.Keys(svcPorts)
+			slices.Sort(svcNames)
+			var ports []string
+			for _, svcName := range svcNames {
+				ports = append(ports, svcPorts[svcName])
+			}
 			return &corev1.EnvVar{
 				Name:  defineKey,
-				Value: strconv.Itoa(int(svc.Spec.Ports[0].Port)),
+				Value: strings.Join(ports, ","),
```

Comment:
The implicit rules here (such as the one-to-one mapping after sorting svcNames) should be explicitly stated in the API documentation.

Reply:
done